float32 left_encoder_val
float32 right_encoder_val